### PR TITLE
fix usdc usdc.e swaps, refactor searchToken to use fuse.js, big,js -> native BigInt

### PR DESCRIPTION
**Pull request by Markeljan**
`enableSmartRouting` is not compatible with all swaps i.e. usdc <-> usdc.e so added a fallback to retry with enableSmartRouting: false.

Tested locally with usdc, usdc.e, near, blackdragon, nearvidia swaps.

### CODE REVIEW

File: src/utils/whitelist-tokens.ts [hunk 1]
```diff
@@ -1,3 +1,14 @@
+export type WhitelistedToken = {
```
Comment (bh2smith) [inline src/utils/whitelist-tokens.ts]: nit: naming conventions
```suggestion
export type AllowlistedToken = {
```

**Review by bh2smith — APPROVED** [review]
Looks good and seems like a great opportunity to change names like whitelist to allowlist!

**Review by sainthiago — APPROVED** [review]

File: src/utils/whitelist-tokens.ts [hunk 2]
```diff
@@ -1,4 +1,15 @@
-export const whitelistedTokens = {
+export type AllowlistedToken = {
```
Comment (bh2smith) [inline src/utils/whitelist-tokens.ts:1]: Thanks so much! Only thing left is the filename...

Comment (Markeljan) [reply src/utils/whitelist-tokens.ts:1]: missed that!

**Review by bh2smith — APPROVED** [review]
Double Approve.

**Review by sainthiago — APPROVED** [review]

File: src/utils/allowlist-tokens.ts [hunk 3]
```diff
@@ -1,4 +1,15 @@
-export const whitelistedTokens = {
+export type AllowlistedToken = {
+    spec?: string;
+    name: string;
+    symbol: string;
+    icon?: string;
+    reference?: string | null;
+    reference_hash?: string | null;
```
Comment (bh2smith) [inline src/utils/allowlist-tokens.ts:7]: Do we explicitly want string or null or undefined? It's our hard coded list, I get we could leave null out here. 
```suggestion
    reference?: string;
    reference_hash?: string;
```

File: src/app/api/[[...slugs]]/route.ts [hunk 4]
```diff
@@ -3,9 +3,6 @@ import { swagger } from '@elysiajs/swagger';
 import { EstimateSwapView, Transaction, WRAP_NEAR_CONTRACT_ID, estimateSwap, fetchAllPools, ftGetTokenMetadata, init_env, instantSwap, nearDepositTransaction, nearWithdrawTransaction, percentLess, scientificNotationToString, separateRoutes } from "@ref-finance/ref-sdk"
 import { Elysia } from "elysia";
 
-// @ts-ignore
-import Big from 'big.js';
```
Comment (bh2smith) [inline src/app/api/[[...slugs]]/route.ts]: Two birds one stone!

File: src/utils/allowlist-tokens.ts (hunk 3 above)
Comment (bh2smith) [reply src/utils/allowlist-tokens.ts:7]: I went and looked at the list. Some of these values are set to null and could maybe be just removed. Some are just empty strings...

**Review by bh2smith — APPROVED** [review]
This looks pretty nice! Love the removal of big. Made a few inline suggestions.

**Review by microchipgnu — APPROVED** [review]
LGTM!

**Comment by Markeljan** [comment]
@microchipgnu added back big.js & rebased with main

File: src/utils/allowlist-tokens.ts (hunk 3 above)
Comment (Markeljan) [reply src/utils/allowlist-tokens.ts:7]: Yes just matching with the types in the JSON, we don't use those fields particularly, just the whole object so leaving as is

File: package.json [hunk 5]
```diff
@@ -19,6 +19,7 @@
     "react-dom": "^18"
   },
   "devDependencies": {
+    "@types/big.js": "^6.2.2",
```
Comment (bh2smith) [inline package.json:22]: but why!?

Comment (microchipgnu) [reply package.json:22]: do we need this?

Comment (Markeljan) [reply package.json:22]: Ref SDK uses it ☹️

Comment (bh2smith) [reply package.json:22]: 🤮

**Review by microchipgnu — APPROVED** [review]

**Comment by Markeljan** [comment]
@microchipgnu added "generate-transaction" primitive and updated MB Assistant spec.

**Pull request merged**